api/src/reconciliation/reconciliation.service.ts

```diff
@@ -95,6 +95,19 @@ export class ReconciliationService {
     });
   }
 
+  async getEnvReconStatusBySHA(org: Organization, gitSha: string) {
+    const recon = await this.envReconRepo.findOne({
```

Comment:
This should return the result of `findOne`. It's up to the controller to decide how it needs to handle responding to the client.

ie: we should not throw `InternalServiceErrorException` or return custom response objects from these service calls. That logic should always be done in the controller.